server/controllers/googleApiController.js

```diff
@@ -66,44 +89,38 @@ function authenticateUser(req, res) {
     .catch((err) => res.status(422).json(err));
 }
 
-function getAvailability(req, res) {
+async function getAvailability(req, res) {
   const { query } = req;
   const date = `${query.year}/${query.month}/${query.date}`; // 2020-08-20
   const meetingLength = parseInt(query.meetingLength); // "30min" => 30
-  const { availableFrom, availableTo, timeZone } = query;
+  const { timeZone, calendarUrl } = query;
+  let user;
 
-  //create time availability range based on user availability preference
-  const availabilityStart = moment(`${date} ${availableFrom}`);
-  const availabilityEnd = moment(`${date} ${availableTo}`);
-  const range = moment.range(availabilityStart, availabilityEnd);
-
-  //slice the range by meetinglength into different slots
-  let timeSlot = Array.from(range.by("minutes", { step: meetingLength }));
-  timeSlot.pop();
+  findUserByUrl(calendarUrl)
+    .then((dbModel) => {
```

Comment:
perhaps put an edge case here incase you can't find any users with that calendarUrl

Reply:
getAvailability is called when someone **opens a calendar page**, and **click on a date** to get this date's availability slot. 
If the user doesn't exist, the calendar page would display"page doesn't exist" in the very first place, and would not allow the user to make getAvailability requests. @ylee297  
So the edge case is taken care of in the client, but as u said, we should double-check on both client and server-side (like available days), I will return a message to the client if the user doesn't exist, instead of time slots.

Reply:
resolved by returning 404 messages if the user doesn't exist ( line 118)